ui/pages/institutional/custody/custody.js

```diff
@@ -208,60 +227,56 @@ const CustodyPage = () => {
 
   useEffect(() => {
     const fetchConnectRequest = async () => {
-      const connectRequestValue = await dispatch(
-        mmiActions.getCustodianConnectRequest(),
-      );
-
-      if (Object.keys(connectRequestValue).length) {
-        const {
-          token,
-          custodianName,
-          custodianType,
-          apiUrl: custodianApiUrl,
-        } = connectRequestValue;
-
-        const jwt =
-          token || (await dispatch(mmiActions.getCustodianToken(address)));
-        const accountsValue = await dispatch(
-          mmiActions.getCustodianAccounts(
-            jwt,
-            custodianApiUrl,
-            custodianType,
-            true,
-          ),
-        );
-
-        setConnectRequest(connectRequestValue);
-        setCurrentJwt(jwt);
-        setSelectedCustodianType(custodianType);
-        setSelectedCustodianName(custodianName);
-        setApiUrl(custodianApiUrl);
-        setConnectError('');
-        setAccounts(accountsValue);
-
-        trackEvent({
-          category: MetaMetricsEventCategory.MMI,
-          event: MetaMetricsEventName.CustodianConnected,
-          properties: {
-            custodian: custodianName,
-            apiUrl,
-            rpc: Boolean(connectRequest),
-          },
-        });
-      }
-    };
-
-    const handleFetchConnectRequest = async () => {
       try {
-        setLoading(true);
-        await fetchConnectRequest();
-        setLoading(false);
+        if (connectRequest && Object.keys(connectRequest).length) {
+          const {
+            token,
+            environment: custodianName,
+            service: custodianType,
+            apiUrl: custodianApiUrl,
+          } = connectRequest;
+
+          const custodianToken =
+            token || (await dispatch(mmiActions.getCustodianToken(address)));
+
+          setCurrentJwt(custodianToken);
+          setSelectedCustodianType(custodianType);
+          setSelectedCustodianName(custodianName);
+          setApiUrl(custodianApiUrl);
+          setConnectError('');
+
+          const accountsValue = await dispatch(
+            mmiActions.getCustodianAccounts(
+              custodianToken,
+              custodianApiUrl,
+              custodianType,
+              true,
+            ),
+          );
+
+          setAccounts(accountsValue);
+
+          trackEvent({
+            category: MetaMetricsEventCategory.MMI,
+            event: MetaMetricsEventName.CustodianConnected,
+            properties: {
+              custodian: custodianName,
+              apiUrl,
+              rpc: Boolean(connectRequest),
+            },
+          });
+        }
       } catch (error) {
         console.error(error);
-        setLoading(false);
+        handleConnectError(error);
       }
     };
 
+    const handleFetchConnectRequest = () => {
```

Comment:
Use the finally method of the Promise returned by the `fetchConnectRequest` function to set the loading state to false after the fetchConnectRequest function has been completed.